include/coruja/container/view/any_container.hpp

```diff
@@ -0,0 +1,139 @@
+
+// Distributed under the Boost Software License, Version 1.0.
+// (See accompanying file LICENSE or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+
+#pragma once
+
+#include <iterator>
+#include <functional>
+#include <memory>
+#include <type_traits>
+#include <utility>
+
+#include "coruja/container/view/container.hpp"
+#include "coruja/support/signal/any_connection.hpp"
+#include "coruja/support/signal/any_signal_id_t.hpp"
+#include "coruja/support/type_traits.hpp"
+#include "range/v3/view/any_view.hpp"
+#include <range/v3/range_traits.hpp>
+
+
+namespace coruja { namespace view {
+
+template<typename T, ranges::category Cat = ranges::category::input>
+class any_container : public ranges::any_view<T, Cat>
+{
+    using base_view = ranges::any_view<T, Cat>;
+public:
+    using iterator = ranges::range_iterator_t<base_view>;
+    using value_type = typename std::decay<T>::type;
+    using for_each_connection_t = any_connection;
+    using before_erase_connection_t = any_connection;
+private:
+
+    struct any_container_iface
+    {
+        virtual ~any_container_iface() = default;
+        virtual any_container_iface* clone() const = 0;
+
+        virtual for_each_connection_t for_each(std::function<void(value_type)>) = 0;
+        virtual for_each_connection_t for_each(std::function<void(any_container&,iterator)>) = 0;
+
+        virtual before_erase_connection_t before_erase(std::function<void(value_type)>) = 0;
+        virtual before_erase_connection_t before_erase(std::function<void(any_container&,iterator)>) = 0;
+    };
+
+    struct wrap_invoker
+    {
+        template<typename From, typename It>
```

Comment:
```suggestion
        template<typename From, typename It, 
        CONCEPT_REQUIRES_(ranges::RandomAccessRange<From>())>
```
I think including this concept here would make it easier to understand any breaches of contracts and compilation errors, but doing so now breaks with the coruja::map.

Reply:
>...but doing so now breaks with the coruja::map

Exactly, it would be a nonsense to have a type erasure like `any_container` that doesn 't accept a container with iterators that doesn't model the `RandomAccessIterator`.